From 8ac89180aec77ec6a8654e7dcf4a2e6c2917e46d Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 21 Nov 2022 19:35:25 +0200
Subject: [PATCH] [SW-105079] gaudi3: pldm: unblock pcie and elbi range register

In gaudi3, HW FSM or ROM blocks the complete config range based
on fuse bit. This patch unblocks RR for PCIE and ELBI using ROM
api only for PLDM plaform. On ASIC, ranges will be unblocked
for required registers which needs to be accessed from PCIe.

Change-Id: Ica72264ae792a9d042a62bd19076ed556f49e3d0
---

diff --git a/arch/gaudi3/arc/pldm.c b/arch/gaudi3/arc/pldm.c
index ba185d2..814a1ff 100644
--- a/arch/gaudi3/arc/pldm.c
+++ b/arch/gaudi3/arc/pldm.c
@@ -10,6 +10,7 @@
 #include <preboot.h>
 #include <uart.h>
 #include <pll.h>
+#include <arcboot_export_dev.h>
 #include "hl.h"
 
 #define SYS_CLOCK_FREQUENCY	250000000ULL /* PCI PLL DIV3 SIMULATION CLK */
@@ -174,3 +175,8 @@
 {
 	hl_pll_init(gaudi3_nw_pll_cfg, ARRAY_SIZE(gaudi3_nw_pll_cfg));
 }
+
+void ppboot_security_init(void)
+{
+	security_rr_open_pcie_config();
+}
